x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/user/InternalUsers.java

```diff
@@ -7,94 +7,73 @@
 
 package org.elasticsearch.xpack.core.security.user;
 
-import org.elasticsearch.core.Nullable;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 import org.elasticsearch.xpack.core.security.authz.RoleDescriptor;
 
 import java.util.HashMap;
 import java.util.Map;
-import java.util.function.Predicate;
 import java.util.stream.Collectors;
 
 public class InternalUsers {
```

Comment:
I wondered whether you'd say this (though it didn't really occur to me until I looked at the diff in GitHub).

If you're happy with a bigger PR, I'm happy to push the refactor further.

Reply:
Yep I am OK with a bigger change along this line. I don't think reservedUser has to be part of it. But that does not seem to add much anyway. Thanks!

Reply:
Yeah, I really only added `ReservedUser` so that all in-code users have an explicit type hierarchy. They're either `ReservedUser` or `InternalUser`, never just a `User`.
It makes it easier to look at a class and know what type of user we're dealing with.